AWSPinpointTests/AWSPinpointNotificationManagerTests.m

```diff
@@ -0,0 +1,153 @@
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import "OCMock.h"
+#import "AWSTestUtility.h"
+#import "AWSPinpoint.h"
+#import "AWSPinpointNotificationManager.h"
+
+@interface AWSPinpointAnalyticsClient()
+@property (nonatomic, strong) NSDictionary* globalFeatureAttributes;
+- (void) setFeatureAttributes:(NSDictionary*) attributes;
+- (void) removeAllGlobalFeatureAttributes;
+@end
+
+@interface AWSPinpointNotificationManager()
+- (void)addPinpointMetadataForEvent:(AWSPinpointEvent *) event
+                   withNotification:(NSDictionary *) userInfo;
+- (void)addGlobalPinpointMetadataForNotification:(NSDictionary *) userInfo;
+- (NSString*)buildEventTypeFromUserInfo:(NSDictionary *) userInfo
+                     withPushActionType:(AWSPinpointPushActionType) pushActionType;
+@end
+
+@interface AWSPinpointConfiguration()
+@property (nonnull, strong) NSUserDefaults *userDefaults;
+@end
+
+@interface AWSPinpointNotificationManagerTests : XCTestCase
+@property (nonatomic, strong) AWSPinpoint *pinpoint;
+@property (nonatomic, strong) NSUserDefaults *userDefaults;
+@property (nonatomic, strong) NSDictionary *campaignPushPayload;
+@property (nonatomic, strong) NSDictionary *journeyPushPayload;
+
+@end
+
+@implementation AWSPinpointNotificationManagerTests
+
+- (void)setUp {
+    [super setUp];
+
+    [[NSUserDefaults standardUserDefaults] removeSuiteNamed:@"AWSPinpointNotificationManagerTests"];
+    self.userDefaults = [[NSUserDefaults alloc] initWithSuiteName:@"AWSPinpointNotificationManagerTests"];
+
+    [AWSTestUtility setupCognitoCredentialsProvider];
+    AWSPinpointConfiguration *configuration = [[AWSPinpointConfiguration alloc] initWithAppId:@"fakeAppId" launchOptions:@{}];
+    configuration.userDefaults = self.userDefaults;
+    configuration.enableAutoSessionRecording = NO;
+    self.pinpoint = [AWSPinpoint pinpointWithConfiguration:configuration];
+
+    _campaignPushPayload = @{
+        @"data": @{
+            @"pinpoint": @{
+                @"campaign": @{
+                    @"campaign_id": @"testCampaignId"
+                }
+            }
+        }
+    };
+
+    _journeyPushPayload = @{
+        @"data": @{
+            @"pinpoint": @{
+                @"journey": @{
+                    @"journey_id": @"testJourneyId"
+                }
+            }
+        }
+    };
+}
+
+- (void)tearDown {
+    [self.pinpoint.analyticsClient.eventRecorder removeAllEvents];
+    [self.pinpoint.analyticsClient removeAllGlobalFeatureAttributes];
+    [super tearDown];
+}
+
+- (void)testAddPinpointMetadataForCampaignEvent {
+    AWSPinpointEvent *pushNotificationEvent = [self.pinpoint.analyticsClient createEventWithEventType:@"_campaign.received_foreground"];
+    [_pinpoint.notificationManager addPinpointMetadataForEvent:pushNotificationEvent
```

Comment:
is there any reason why we are using the iVar here and not `self.pinpoint` ? (along with everywhere else in this file)

Reply:
Will update